gprofiler/merge.py

```diff
@@ -332,6 +333,46 @@ def _enrich_and_finalize_stack(
     return f"{enrich_data.application_prefix}{enrich_data.container_prefix}{stack} {count}"
 
 
+def _merge_metadata(primary, secondary):
+    for key, value in primary.items():
+        # in case value is a dict itself
+        if isinstance(value, dict):
+            node = secondary.setdefault(key, {})
+            _merge_metadata(value, node)
+        else:
+            secondary[key] = value
+    return secondary
+
+
+def concatenate_from_external_file(
+    collapsed_file_path: str,
+    obtained_metadata: Metadata,
+) -> Tuple[str, str, str, int]:
+    """
+    Concatenate all stacks from all stack mappings in process_profiles.
+    Add "profile metadata" and metrics as the first line of the resulting collapsed file.
+    """
+    total_samples = 0
+    lines = []
+    start_time = None
+    end_time = None
+
+    # TODO: container names and application metadata
+    with open(collapsed_file_path) as file:
+        for index, line in enumerate(file):
+            if index == 0:
+                read_metadata = json.loads(line[2:])
+                metadata = _merge_metadata(read_metadata, obtained_metadata)
+                try:
+                    start_time = datetime.datetime.strptime(metadata["start_time"], "%Y-%m-%dT%H:%M:%S.%f")
```

Comment:
Let's move parsing to a utility function `parse_iso8601_timestamp`? Place it next to `get_iso8601_format_time`.